icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -1705,7 +1705,34 @@ class RemapRule : public SegmentationRule {
                 resolved[i].appliedRule = this;
                 resolved[i].indexInRemapped.reset();
             }
+            // While replacing, we need to check that we are not creating
+            // surrogate pairs.  Since appendReplacement performs two
```

Comment:
Where do unpaired surrogates come from in the first place?
Rather than check that two unpaired ones don't make a pair, shouldn't we take care not to write unpaired surrogates?

Reply:
The monkeys intentionally test that things work consistently with unpaired surrogates (they go to some effort to avoid generating pairs while randomly generating surrogates).

Note that the line breaking algorithm has a dedicated category for surrogates.

It probably makes sense to test that things don’t horribly break with those anyway, instead of waiting for the fuzzer to find things.

We have an option to suppress them, which is used when testing ICU4X (where strings cannot contain unpaired surrogates).